web/src/main/java/org/mskcc/cbio/portal/web/ProxySessionServiceController.java

```diff
@@ -110,149 +114,220 @@ public HttpHeaders getHttpHeaders() {
         }};
      }
     
-    @RequestMapping(value = "/virtual_study", method = RequestMethod.GET)
-    public @ResponseBody List<VirtualStudy> getUserStudies() {
-        List<VirtualStudy> virtualStudies = new ArrayList<>();
-        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
-        if (!StringUtils.isEmpty(sessionServiceURL) && authentication != null && !(authentication instanceof AnonymousAuthenticationToken)) {
-            RestTemplate restTemplate = new RestTemplate();
-            HttpEntity<Object> headers = new HttpEntity<Object>(getHttpHeaders());
-            ResponseEntity<VirtualStudy[]> responseEntity = restTemplate.exchange(
-                    sessionServiceURL + "virtual_study/query?field=data.users&value=" + 
-                    ((UserDetails)authentication.getPrincipal()).getUsername(),
-                    HttpMethod.GET,
-                    headers,
-                    VirtualStudy[].class);
-            virtualStudies = Arrays.asList(responseEntity.getBody());
-        }
-        return virtualStudies;
-        
-    }
+	@RequestMapping(value = "/virtual_study", method = RequestMethod.GET)
+	public @ResponseBody List<VirtualStudy> getUserStudies() throws JsonProcessingException {
+		List<VirtualStudy> virtualStudies = new ArrayList<>();
+		Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+		if (isSessionServiceEnabled()
+				&& !(authentication == null || authentication instanceof AnonymousAuthenticationToken)) {
+
+			Map<String, String> map = new HashMap<>();
+			map.put("data.users", ((UserDetails) authentication.getPrincipal()).getUsername());
+
+			ObjectMapper mapper = new ObjectMapper();
+			String virtualStudyStr = mapper.writeValueAsString(map);
+
+			RestTemplate restTemplate = new RestTemplate();
+
+			HttpEntity<Object> httpEntity = new HttpEntity<Object>(virtualStudyStr, getHttpHeaders());
+
+			ResponseEntity<List<VirtualStudy>> responseEntity = restTemplate.exchange(
```

Comment:
@kalletlak is it necessary to use the new `POST` endpoint here? Would again be nice if we could be backwards compatible. Only use the new endpoint if necessary. I guess when the type is `group`?

Reply:
its not. I'll revert it back